Update guides according to `lsp-smart-contracts v0.8.0` (#462)
lukso-network/docs: docs/guides/universal-profile/transfer-lyx.md

const myKM = new ethers.Contract(owner, KeyManager.abi, provider);
```

  </TabItem>

</Tabs>

## Step 3 - Encode the calldata to transfer LYX

With our contract instances ready, we now want to transfer some LYX from the UP using the `execute(...)` function.
The next step is to ABI encode this function call with the correct parameters, as explained in the introduction.

We can use the [`encodeABI()`](https://web3js.readthedocs.io/en/v1.7.4/web3-eth-contract.html#methods-mymethod-encodeabi) method from web3.js

<Tabs>
  
  <TabItem value="web3js" label="web3.js">

```typescript
const OPERATION_CALL = 0;
const recipient = '0x...'; // address the recipient (any address, including an other UP)
const amount = web3.utils.toWei('3'); // amount of LYX we want to transfer
// calldata executed at the target (here nothing, just a plain LYX transfer)
const data = '0x';

// encode the calldata to transfer 3 LYX from the UP
const transferLYXCalldata = await myUP.methods[
  'execute(uint256,address,uint256,bytes)'
](OPERATION_CALL, recipient, amount, data).encodeABI();
```

  </TabItem>

  <TabItem value="ethersjs" label="ethers.js">

```typescript
const OPERATION_CALL = 0;

> const recipient = '0x...'; // address the recipient (any address, including an other UP)

Reviewer: ```suggestion
const recipient = '0x...'; // address of the recipient (any address, including an other UP)
```

const amount = ethers.parseEther('3'); // amount of LYX we want to transfer

> // calldata executed at the target (here nothing, just a plain LYX transfer)

Reviewer: you can make this inline with the data variable to stay consistent

const PRIVATE_KEY = '0x...'; // your EOA private key (controller address)

const myEOA = new ethers.Wallet(PRIVATE_KEY).connect(provider);
```

  </TabItem>

</Tabs>

### 4.2 - Send the LYX transfer calldata

The final step is to pass the encoded LYX transfer calldata to the Key Manager. Since we are calling from a UP's controller address (with proper [**permissions**](../../standards/universal-profile/lsp6-key-manager.md#permissions)), the Key Manager will authorize and execute the LYX transfer.

<Tabs>
  
  <TabItem value="web3js" label="web3.js">

```typescript
await myKM.methods['execute(bytes)'](transferLYXCalldata).send({
  from: myEOA.address,
  gasLimit: 300_000,
});
```

  </TabItem>

  <TabItem value="ethersjs" label="ethers.js">

```typescript
await myKM.connect(myEOA)['execute(bytes)'](transferLYXCalldata);
```

  </TabItem>

</Tabs>

## Final Code

<Tabs>
  
  <TabItem value="web3js" label="web3.js">

```javascript
import UniversalProfile from '@lukso/lsp-smart-contracts/artifacts/UniversalProfile.json';
import KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json';
import Web3 from 'web3';

const web3 = new Web3('https://rpc.l16.lukso.network');

> const PRIVATE_KEY = '0x...'; // your EOA private key (controller address)

Reviewer: ```suggestion
const PRIVATE_KEY = '0x...'; // your EOA controller address private key
```

const OPERATION_CALL = 0;
const recipient = '0x...'; // address the recipient (any address, including an other UP)
const amount = web3.utils.toWei('3');
// calldata executed at the target (here nothing, just a plain LYX transfer)
const data = '0x';

// 2. encode the calldata to transfer 3 LYX from the UP
const transferLYXCalldata = await myUP.methods[
  'execute(uint256,address,uint256,bytes)'
](OPERATION_CALL, recipient, amount, data).encodeABI();

// 3. execute the LYX transfer via the Key Manager
await myKM.methods['execute(bytes)'](transferLYXCalldata).send({
  from: myEOA.address,
  gasLimit: 300_000,
});
```

  </TabItem>

  <TabItem value="ethersjs" label="ethers.js">

```typescript
import UniversalProfile from '@lukso/lsp-smart-contracts/artifacts/UniversalProfile.json';
import KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json';
import { ethers } from 'ethers';

const provider = new ethers.JsonRpcProvider('https://rpc.l14.lukso.network');

const myUPAddress = '0x...';
const myUP = new ethers.Contract(myUPAddress, UniversalProfile.abi, provider);

// the KeyManager is the owner of the Universal Profile

> // so we can call the owner() function to obtain the KeyManager's address

Reviewer: ```suggestion
// so we can call the owner() function to obtain the KeyManager contract address
```
